### PR TITLE
fix: xml formatting when moving element

### PR DESCRIPTION
**Pull request by nahmed22**
Fix  readability issue with formatting in `annotation.xml` after section is droped/moved to group section

### CODE REVIEW

File: packages/fiori-annotation-api/src/xml/writer.ts [hunk 1]
```diff
@@ -268,6 +268,40 @@ function handleXmlElementAttributeInserts(element: XMLElement, attributeInserts:
     return edits;
 }
 
+function getTextFragmentIndentation(text: string): number {
+    const lines = text.split('\n');
+    const linesWithTextIndices = lines.map((line, idx) => (line.length > 0 ? idx : -1));
+    const indentTextIndex = linesWithTextIndices.find((idx) => idx > -1);
+    if (indentTextIndex == undefined) {
```
Comment (vadson71) [inline packages/fiori-annotation-api/src/xml/writer.ts:275]: === are better than ==

File: packages/fiori-annotation-api/src/xml/writer.ts [hunk 2]
```diff
@@ -268,6 +268,40 @@ function handleXmlElementAttributeInserts(element: XMLElement, attributeInserts:
...
+        return 0;
+    }
+    const indentLineText = lines[indentTextIndex];
+    return Math.ceil(indentLineText.length - indentLineText.trimStart().length) / 4;
```
Comment (vadson71) [inline packages/fiori-annotation-api/src/xml/writer.ts:279]: ceil should be calculated after division by 4
Would be nice to have a test case for it

**Review by vadson71 — CHANGES_REQUESTED** [review]

File: packages/fiori-annotation-api/src/xml/writer.ts (hunk 1 above)
Comment (nahmed22) [reply packages/fiori-annotation-api/src/xml/writer.ts]: done

File: packages/fiori-annotation-api/src/xml/writer.ts (hunk 2 above)
Comment (nahmed22) [reply packages/fiori-annotation-api/src/xml/writer.ts]: done

**Review by vadson71 — APPROVED** [review]
Changes look good, Comments addressed. Tests coverage is good

**Pull request merged**